src/create-element.js

```diff
@@ -10,9 +10,19 @@ import options from './options';
  */
 export function createElement(type, props, children) {
 	let normalizedProps = {},
-		i;
+		i,
+		ref;
+
 	for (i in props) {
-		if (i !== 'key' && i !== 'ref') normalizedProps[i] = props[i];
+		if (i == 'ref') {
+			if (type && type._forwarded) {
```

Comment:
See comment in `forwardRef.js` for why this was moved to core

Reply:
I have this exact same code in a local branch
So i could use the same mechanism on the composition API
And get rid of options.vnode also

Reply:
I kinda dislike the notion of mixing concerns, preact-core has no notion of forwarding props so I kinda dislike putting that here, if I'm the only one however feel free to ignore that

Reply:
Yea, I agree that I'd prefer this live in compat but the byte savings were too good to pass up I thought lol

Reply:
I'm really a fan of these savings, but I'm with @JoviDeCroock here. For as long we have the `core` vs `compat` separation it feels wrong to put code into `core` that is only needed for `compat`.